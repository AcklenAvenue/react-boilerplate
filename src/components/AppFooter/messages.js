import { defineMessages } from 'react-intl';

const messages = defineMessages({
  BackendVersion: {
    id: 'App.Footer.BackendVersion',
    defaultMessage: 'Back-End Version (get from the api)'
  },
  FrontEndInformation: {
    id: 'App.Footer.FrontEndInformation',
    defaultMessage: 'Front-End Version {deploymentEnvironment} {frontEndBuildNumber}'
  }
});

export default messages;
