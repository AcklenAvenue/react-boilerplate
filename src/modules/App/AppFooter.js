// @flow
import type {AppProps} from './types';
import React, {Component} from 'react';
import {Well} from 'react-bootstrap';

class AppFooter extends Component<AppProps> {
  render() {
    return (
      <Well>
        <p>
          Front-End Version {
            this.props.buildInformation.deploymentEnvironment
          }{' '}
          {this.props.buildInformation.frontEndBuldNumber}
        </p>
        <p>Back-End Version (get from the api) </p>
      </Well>
    );
  }
}

export default AppFooter;
