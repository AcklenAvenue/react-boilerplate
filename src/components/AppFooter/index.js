// @flow
import type { BuildInformation } from '../../containers/App/types';
import React from 'react';
import { Well } from 'react-bootstrap';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const AppFooter = ({
  deploymentEnvironment,
  frontEndBuildNumber
}: BuildInformation) => (
  <Well>
    <p>
      <FormattedMessage
        {...messages.FrontEndInformation}
        values={{ deploymentEnvironment, frontEndBuildNumber }}
      />
    </p>
    <p>
      {' '}
      <FormattedMessage {...messages.BackendVersion} />{' '}
    </p>
  </Well>
);

export default AppFooter;
