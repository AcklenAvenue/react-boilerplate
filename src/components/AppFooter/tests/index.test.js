import React from 'react';
import AppFooter from '../index';
import renderer from 'react-test-renderer';

describe('<App />', () => {
  it('Expect to be render correctly', () => {
    const props = {
      deploymentEnvironment: 'test',
      frontEndBuildNumber: 123
    };
    const wrapper = renderer.create(<AppFooter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
