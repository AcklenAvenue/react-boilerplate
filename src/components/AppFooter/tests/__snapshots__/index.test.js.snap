// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppFooter /> Expect to be render correctly 1`] = `
<div
  className="well"
>
  <p>
    <span>
      Front-End Version test 123
    </span>
  </p>
  <p>
     
    <span>
      Back-End Version (get from the api)
    </span>
     
  </p>
</div>
`;
